data-products/src/main/scala/org/sunbird/obsrv/dataproducts/job/MasterDataProcessorIndexer.scala

```diff
@@ -0,0 +1,156 @@
+package org.sunbird.obsrv.dataproducts.job
+
+import com.redislabs.provider.redis._
+import com.typesafe.config.{Config, ConfigFactory}
+import kong.unirest.Unirest
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.{SparkConf, SparkContext}
+import org.joda.time.format.{DateTimeFormat, DateTimeFormatter}
+import org.joda.time.{DateTime, DateTimeZone}
+import org.json4s.native.JsonMethods._
+import org.sunbird.obsrv.core.util.JSONUtil
+import org.sunbird.obsrv.dataproducts.helper.{BaseMetricHelper, KafkaMessageProducer}
+import org.sunbird.obsrv.dataproducts.model.{Edata, MetricLabel}
+import org.sunbird.obsrv.model.DatasetModels.{DataSource, Dataset}
+import org.sunbird.obsrv.registry.DatasetRegistry
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+
+import scala.collection.mutable
+
+object MasterDataProcessorIndexer {
+  private final val logger: Logger = LogManager.getLogger(MasterDataProcessorIndexer.getClass)
+  private val config: Config = ConfigFactory.load("masterdata-indexer.conf").withFallback(ConfigFactory.systemEnvironment())
+  private val dayPeriodFormat: DateTimeFormatter = DateTimeFormat.forPattern("yyyyMMdd").withZoneUTC()
+
+  case class Paths(datasourceRef: String, ingestionPath: String, outputFilePath: String, timestamp: Long)
+  case class BlobProvider(sparkProviderURIFormat: String, druidProvider: String, druidProviderPrefix: String)
+
+  def main(args: Array[String]): Unit = {
+    val datasets = DatasetRegistry.getAllDatasets("master-dataset")
+    val indexedDatasets = datasets.filter(dataset => {
+      dataset.datasetConfig.indexData.nonEmpty && dataset.datasetConfig.indexData.get
+    })
+    val metrics = BaseMetricHelper(config)
+    indexedDatasets.foreach(dataset => {
+      metrics.generate(ets = new DateTime(DateTimeZone.UTC).getMillis, datasetId = dataset.id, edata = Edata(metric = Map(metrics.getMetricName("total_dataset_count") -> 1), labels = List(MetricLabel("job", "MasterDataIndexer"), MetricLabel("datasetId", dataset.id), MetricLabel("cloud", s"${config.getString("cloudStorage.provider")}"))))
+        indexDataset(dataset, metrics, System.currentTimeMillis())
+    })
+  }
+
+  def indexDataset(dataset: Dataset, metrics: BaseMetricHelper, time: Long): Unit = {
+    try {
+      val datasources = DatasetRegistry.getDatasources(dataset.id)
+      if (datasources.isEmpty || datasources.get.size > 1) {
```

Comment:
Why are we returning if the data sources are already defined?

Reply:
Currently there are no rollups, so that can be removed.